sds_data_manager/lambda_code/SDSCode/database_handler.py

```diff
@@ -29,58 +29,55 @@ def update_status_table(status_params):
     version = status_params["version"]
 
     try:
-        with Session(db.get_engine()) as session:
-            # Had to query this way because the select statement
-            # returns a RowProxy object when it executes it,
-            # not the actual StatusTracking model instance,
-            # which is why it can't update table row directly.
-            result = (
-                session.query(models.StatusTracking)
-                .filter(models.StatusTracking.instrument == instrument)
-                .filter(models.StatusTracking.data_level == data_level)
-                .filter(models.StatusTracking.descriptor == descriptor)
-                .filter(models.StatusTracking.start_date == start_date)
-                .filter(models.StatusTracking.version == version)
-                .first()
-            )
+        # Had to query this way because the select statement
+        # returns a RowProxy object when it executes it,
+        # not the actual StatusTracking model instance,
+        # which is why it can't update table row directly.
+        result = (
+            session.query(models.StatusTracking)
+            .filter(models.StatusTracking.instrument == instrument)
+            .filter(models.StatusTracking.data_level == data_level)
+            .filter(models.StatusTracking.descriptor == descriptor)
+            .filter(models.StatusTracking.start_date == start_date)
+            .filter(models.StatusTracking.version == version)
+            .first()
+        )
 
-            if result is None:
-                logger.info(
-                    "No existing record found, creating"
-                    f" new record for {instrument}, {data_level},"
-                    f"{descriptor}, {start_date}, {version}"
-                )
-                session.add(models.StatusTracking(**status_params))
-                session.commit()
-            else:
-                logger.info(
-                    f"Updating existing record, {result.__dict__}, with batch info"
-                )
-                result.status = status_params["status"]
-                result.job_definition = status_params["job_definition"]
-                result.job_log_stream_id = status_params["job_log_stream_id"]
-                result.container_image = status_params["container_image"]
-                result.container_command = status_params["container_command"]
-                session.commit()
+        if result is None:
+            logger.info(
+                "No existing record found, creating"
+                f" new record for {instrument}, {data_level},"
+                f"{descriptor}, {start_date}, {version}"
+            )
+            session.add(models.StatusTracking(**status_params))
+            session.commit()
+        else:
+            logger.info(f"Updating existing record, {result.__dict__}, with batch info")
```

Comment:
It seems like if the record is found, it doesn't attempt to write the instrument, version, etc, meaning we won't hit the unique constraint if the record already exists in the db...?

Reply:
Agreed, you are asking great questions here! This comes in the next PR.